src/Bicep.Core/Diagnostics/DiagnosticBuilder.cs

```diff
@@ -406,6 +404,11 @@ public ErrorDiagnostic ArgumentCountMismatch(int argumentCount, int mininumArgum
                 TextSpan,
                 "BCP078",
                 $"The property '{propertyName}' requires a value of type {expectedType}, but none was supplied.");
+
+            public ErrorDiagnostic CyclicSelfReference() => new ErrorDiagnostic(
+                TextSpan,
+                "BCP079",
+                "This expression is referencing its own declaration in a cycle.");
```

Comment:
>This expression is referencing its own declaration in a cycle. [](start = 17, length = 62)

I think we should change this one to "This expression is referencing its own declaration, which is not allowed." (Can't self-reference without it being a cycle.)